processor/k8sattributesprocessor/internal/kube/client.go

```diff
@@ -370,12 +371,20 @@ func (c *WatchClient) extractPodContainersAttributes(pod *api_v1.Pod) map[string
 	if c.Rules.ContainerImageName || c.Rules.ContainerImageTag {
 		for _, spec := range append(pod.Spec.Containers, pod.Spec.InitContainers...) {
 			container := &Container{}
-			imageParts := strings.Split(spec.Image, ":")
-			if c.Rules.ContainerImageName {
-				container.ImageName = imageParts[0]
+			ref, err := reference.Parse(spec.Image)
+			if err != nil {
+				c.logger.Warn("Invalid image reference found", zap.String("image", spec.Image), zap.Error(err))
+				continue
+			}
+
+			if named, ok := ref.(reference.Named); c.Rules.ContainerImageName && ok {
+				container.ImageName = named.Name()
+			}
+			if tagged, ok := ref.(reference.Tagged); c.Rules.ContainerImageTag && ok {
+				container.ImageTag = tagged.Tag()
 			}
-			if c.Rules.ContainerImageTag && len(imageParts) > 1 {
-				container.ImageTag = imageParts[1]
+			if digested, ok := ref.(reference.Digested); c.Rules.ContainerImageTag && ok {
+				container.ImageTag = digested.Digest().String()
```

Comment:
Why do we want to set digest as an image tag? Not sure if `container.image.tag` can have digest as its value. If you think otherwise we should probably mention this in the spec https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/resource/semantic_conventions/container.md